File: app/src/main/java/com/demo/android_view/activity/MainActivity.java
package com.demo.android_view.activity;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import com.demo.android_view.R;
import com.demo.android_view.mview.RingProgressBarView;

/**
 * created by tea9 at 2018/11/15
 */
public class MainActivity extends AppCompatActivity {
    private float mTotalProgress = 40f;
    private float mCurrentProgress = 0;
    //进度条
    private RingProgressBarView mTasksView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mTasksView = (RingProgressBarView) findViewById(R.id.tasks_view);
        mTasksView.setProgress(43.9f);

//        new Thread(new ProgressRunable()).start();
    }

    class ProgressRunable implements Runnable {
        @Override
        public void run() {
            while (mCurrentProgress < mTotalProgress+1) {
                mCurrentProgress += 1;
                if (mCurrentProgress>mTotalProgress){
                    mCurrentProgress = mTotalProgress;
                }
                mTasksView.setProgress(mCurrentProgress);
                try {
                    Thread.sleep( 57);
                } catch (Exception e) {
                    e.printStackTrace();
                }
            }
        }
    }

}

File: app/src/main/java/com/demo/android_view/MainActivity.kt
package com.demo.android_view

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import com.demo.android_view.activity.*
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(), View.OnClickListener {
    override fun onClick(p0: View?) {
        when(p0!!.id){
            R.id.btn1 -> startActivity(Intent(this,LineProgressBarActivity::class.java))
            R.id.btn2 -> startActivity(Intent(this,RingProgressBarActivity::class.java))
            R.id.btn3 -> startActivity(Intent(this,VerticalBarViewActivity::class.java))
            R.id.btn4 -> startActivity(Intent(this,LineProgressOActivity::class.java))
            R.id.btn5 -> startActivity(Intent(this,PolygonActivity::class.java))
            R.id.btn6 -> startActivity(Intent(this,PolygonViewActivity::class.java))
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
       setContentView(R.layout.activity_main)

        btn1.setOnClickListener(this)
        btn2.setOnClickListener(this)
        btn3.setOnClickListener(this)
        btn4.setOnClickListener(this)
        btn5.setOnClickListener(this)
        btn6.setOnClickListener(this)
    }
}
